api/controllers/userController.js

```diff
@@ -30,21 +55,109 @@ function usersPut(req, res) {
             birthDate: Date.parse(birthDate)
           })
           .then(() => {
-            res.json({'message': 'OK'});
+            res.json({'message': 'OK'}, 200);
           })
           .catch(err => {
             console.log(err);
             res.json({'message': 'ERROR'}, 500);
           });
         } else {
-          res.json({'message': 'User not found'}, 400);
+          res.json({'message': 'User not found'}, 401);
         }
       });
   } else {
-    res.json({'message': 'Wrong date'}, 400);
+    res.json({'message': 'Wrong date'}, 401);
   }
 }
+  
+function usersLogin(req, res) {
+  var email    = req.swagger.params.userCredentials.value.email;
+  var password = req.swagger.params.userCredentials.value.password;
+  if (email && password) {
+    db.users.findOne({ where: {email: email} })
+      .then(user => {
+        if (user) {
+          bcrypt.compare(password, user.hashword)
+            .then(result => {
+             if (result) {
+              //Create token
+              var token = uuidV4();
+              //Store token in auth_token table
 
+              //Return token
+              res.status(200).json({'token': `${token}`});
+            } else {
+              console.log('ERROR (email or) password is wrong');
+              res.json({'message': 'ERROR (email or) password is wrong'}, 401);
+            }
+          });
+        } else {
+          res.json({'message': 'User not found'}, 401);
+          //TODO go to /signup instead?
+        }
+    });
+  } else {
+    res.json({'message': 'ERROR email are password are required'}, 401);
+  }
+}
+
+function usersSignup(req, res) {
+  var firstName  = req.swagger.params.userInfo.value.firstName;
+  var lastName   = req.swagger.params.userInfo.value.lastName;
+  var birthMonth = req.swagger.params.userInfo.value.birthMonth;
+  var birthDay   = req.swagger.params.userInfo.value.birthDay;
+  var birthYear  = req.swagger.params.userInfo.value.birthYear;
+  var birthDate  = util.format('%d-%d-%d,', birthYear, birthMonth, birthDay);
+  var email      = req.swagger.params.userInfo.value.email;
+  var password   = req.swagger.params.userInfo.value.password;
+  //TODO add other profile fields
+
+  if (email && password && ((birthDay>0 && birthDay<32) && (birthMonth>0 && birthMonth<13) && birthYear)) {
+    //Create hash from password
+    bcrypt.hash(password, 10)
+      .then(hash => {
+        //Create user and store hash
+        db.users.create({
```

Comment:
You can use `findOrCreate` here and return an error message saying that the user already exists if `findOrCreate` does not create a new user. See the pattern here: https://stackoverflow.com/a/32109687 Returning a `400` (bad request) is probably appropriate here.
You can add this functionality in a subsequent PR.